server/server/views/user.py

```diff
@@ -32,8 +32,27 @@ def get(self):
     def put(self):
         request = self.request
         data = request.validated
-        
+        data_is_updated = True
+        update_status = False
+
+        check_dict = {
+            "first_name",
+            "last_name",
+            "nickname",
+            "location"
+        }
+
+        response = {'success': False}
+
+        if not any(data.get(key) for key in check_dict):
+            data_is_updated = False
```

Comment:
I really have no idea what are you checking here. Looks like if any of listed fields or all of them are present in `data` and is not None, you will not update DB. I'm not sure it is expected behavior?